# GEOSPATIAL DATA: SOLAR FARM IN MALAYSIA

## Introduction

A solar farm, also known as a solar power plant or solar array, is a large-scale installation of solar panels designed to generate electricity from sunlight. It is a key component of renewable energy infrastructure that harnesses the abundant energy provided by the sun to produce clean and sustainable power. Solar farms play a crucial role in the global shift toward reducing greenhouse gas emissions and transitioning to more environmentally friendly energy sources.

Solar farms consist of an array of photovoltaic (PV) panels that convert sunlight directly into electricity using the photovoltaic effect. These panels are typically mounted on ground-mounted structures or elevated structures, such as solar trackers, to optimize their exposure to sunlight throughout the day. The generated electricity is then either fed directly into the local power grid or used for on-site consumption.

Advancements in solar panel technology, manufacturing processes and economies of scale have significantly lowered the cost of solar energy production over the years. As a result, solar farms have become increasingly viable and competitive sources of electricity, contributing to the diversification of energy sources and helping to achieve energy sustainability and energy independence goals.

Malaysia possesses abundant renewable energy (RE) assets, with its equatorial position making solar energy a prominent option. The country experiences a monthly solar irradiation of approximately 4000-5000 Wh/m². Furthermore, there is an average of 4 to 8 hours of daily sunshine each month, resulting in a yearly total of about 2200 hours of sunlight. This equates to a substantial energy potential of 8,800-11,000 kWh, achievable with a compact collection area comparable to 16 sheets of A4 paper joined together.

Below is the descriptions of column labels in Malaysia's Solar Farm dataset:

<br>
<img src = "Figure1.png", style="width: 525px";>

## Statement

### Background
Malaysia is committed to transitioning to renewable energy sources, with solar energy playing a crucial role in achieving sustainability targets. However, the expansion of solar panel installations often requires deforestation, which can lead to adverse environmental consequences, including loss of biodiversity, disruption of ecosystems and reduced carbon sequestration.

### Problem Statement
Balancing Solar Panel Development and Forest Conservation for Sustainable Energy Transition in Malaysia.<br>
This project aims to quantify the environmental and economic impact of deforestation for solar panel development in Malaysia and propose a comprehensive compensation framework to ensure the long-term sustainability of the ecosystem.

### Keywords
- Deforestation: The process of clearing forests or wooded areas, often for purposes such as land development or infrastructure expansion, which may lead to habitat loss and environmental disruption
- Renewable Energy: Energy derived from sources that are naturally replenished, such as sunlight (solar energy), wind, and water, aiming to reduce reliance on fossil fuels.
- Sustainable Energy Transition: The shift from non-renewable energy sources (e.g., fossil fuels) to renewable sources (e.g., solar, wind) in a way that meets present needs without compromising the ability of future generations to meet their own needs.
- Solar Farm: A large-scale installation of solar panels used to capture sunlight and convert it into electricity for power generation.
- Long-Term Sustainability: The consideration of how actions taken today, such as deforestation for solar panels, will impact the environment, society, and the economy over extended periods, emphasizing responsible resource use.
- Net Zero Carbon Emissions: Achieving a balance between the amount of greenhouse gases emitted into the atmosphere and the amount removed, resulting in no net contribution to global warming and climate change

## Visualisation of Ploygon Maps

In [ ]:
import pandas as pd
import geopandas as gpd
import folium

In [ ]:
df = pd.read_csv('Solar_Farm_Malaysia_v0.2.0_Polygon - Sheet1.csv')

In [ ]:
df.head()

name  poi_id  \
0  Large Scale Solar (LSS) Power Plant Pekan (100...     NaN   
1  Large Scale Solar (LSS) Power Kuala Perlis - C...     NaN   
2  Large Scale Solar (LSS) Power Plant Padang Bes...     NaN   
3  Large Scale Solar (LSS) Power Plant Arau (5.2 MW)     NaN   
4  Large Scale Solar (LSS) Power Plant Ayer Keroh...     NaN   

                  category                                            address  \
0  Large Scale Solar (LSS)  41, Jln Kampung Tanjung Selangor, 26600 Pekan,...   
1  Large Scale Solar (LSS)                         02000 Kuala Perlis, Perlis   
2  Large Scale Solar (LSS)            Titi Tinggi, 02100 Padang Besar, Perlis   
3  Large Scale Solar (LSS)            Unnamed Road, 02500, 02500 Arau, Perlis   
4  Large Scale Solar (LSS)  Kompleks Hijau Solar, Jalan Solar 1, Hang Tuah...   

    country   state  city      district  postcode  street_building_name  ...  \
0  Malaysia  Pahang   NaN         Pekan       NaN                   NaN  ...   
1  Malaysia  Perlis   NaN  Kuala Perlis       NaN                   NaN  ...   
2  Malaysia  Perlis   NaN  Padang Besar       NaN                   NaN  ...   
3  Malaysia  Perlis   NaN          Arau       NaN                   NaN  ...   
4  Malaysia  Melaka   NaN   Ayer Keroh        NaN                   NaN  ...   

                                         data_source  polygon_area_m2  \
0                                                NaN     1.054558e+06   
1                                                NaN     8.447737e+04   
2                                                NaN     5.698563e+04   
3  https://www.plusxnergy.com/solutions/energy-ge...     4.600205e+04   
4                                                NaN     6.267948e+04   

   polygon_area_ha co2_storage_t  capacity_kW  annual_generation_kW  \
0       105.455764   1160.013400     100000.0            29574000.0   
1         8.447737     92.925112      48000.0            14195520.0   
2         5.698563     62.684189       5000.0             1478700.0   
3         4.600205     50.602257       5200.0             1537848.0   
4         6.267948     68.947423       8000.0             2365920.0   

   co2_without_solar_panels_kg co2_with_solar_panels_kg  co2_reduction_kg  \
0                 1.500600e+11             7.501521e+09      1.425585e+11   
1                 1.500600e+11             7.502290e+09      1.425577e+11   
2                 1.500600e+11             7.502926e+09      1.425571e+11   
3                 1.500600e+11             7.502923e+09      1.425571e+11   
4                 1.500600e+11             7.502882e+09      1.425571e+11   

  co2_reduction_t  
0    1.425585e+08  
1    1.425577e+08  
2    1.425571e+08  
3    1.425571e+08  
4    1.425571e+08  

[5 rows x 36 columns]

In [ ]:
df.shape

(33, 36)

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 33 entries, 0 to 32
Data columns (total 36 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   name                            33 non-null     object 
 1   poi_id                          0 non-null      float64
 2   category                        33 non-null     object 
 3   address                         32 non-null     object 
 4   country                         33 non-null     object 
 5   state                           33 non-null     object 
 6   city                            0 non-null      float64
 7   district                        33 non-null     object 
 8   postcode                        0 non-null      float64
 9   street_building_name            0 non-null      float64
 10  unit_room_number                0 non-null      float64
 11  parlimen                        0 non-null      float64
 12  dun                             0 non-

In [ ]:
df.columns

Index(['name', 'poi_id', 'category', 'address', 'country', 'state', 'city',
       'district', 'postcode', 'street_building_name', 'unit_room_number',
       'parlimen', 'dun', 'coordinate', 'longitude', 'latitude', 'geometry',
       'polygon_farm_footprint_geojson', 'capacity_MW', 'operator_company',
       'url_operator', 'developer_company', 'url_developer',
       'date_commisioned', 'date_completed', 'operation_status', 'data_source',
       'polygon_area_m2', 'polygon_area_ha', 'co2_storage_t', 'capacity_kW',
       'annual_generation_kW', 'co2_without_solar_panels_kg',
       'co2_with_solar_panels_kg', 'co2_reduction_kg', 'co2_reduction_t'],
      dtype='object')

In [ ]:
df.isna().sum()

name                               0
poi_id                            33
category                           0
address                            1
country                            0
state                              0
city                              33
district                           0
postcode                          33
street_building_name              33
unit_room_number                  33
parlimen                          33
dun                               33
coordinate                         0
longitude                          0
latitude                           0
geometry                          33
polygon_farm_footprint_geojson     0
capacity_MW                        0
operator_company                  26
url_operator                      30
developer_company                  0
url_developer                     11
date_commisioned                  27
date_completed                    28
operation_status                   4
data_source                       27
p

In [ ]:
df.dropna(subset=['polygon_farm_footprint_geojson'], inplace=True)

### Visualisation of solar farm on map

In [ ]:
# Calculate map center based on average latitude and longitude
map_center = (df['latitude'].mean(), df['longitude'].mean())
map_object = folium.Map(location=map_center, zoom_start=7)

# Iterate through the dataframe and add GeoJSON data to the map
for index, row in df.iterrows():
    geojson_data = row["polygon_farm_footprint_geojson"]
    folium.GeoJson(geojson_data).add_to(map_object)

# Add tile layer to map
tile = folium.TileLayer(
      tiles = 'https://server.arcgisonline.com/ArcGIS/rest/services/World_Imagery/MapServer/tile/{z}/{y}/{x}',
      attr = 'Esri',
      name = 'Esri Satellite',
      overlay = False,
      control = True
      ).add_to(map_object)

# Display the map
map_object

#### Insert map marker with popup info

In [ ]:
# Calculate map center based on average latitude and longitude
map_center = (df['latitude'].mean(), df['longitude'].mean())
map_object = folium.Map(location=map_center, zoom_start=7)

# Iterate through the dataframe and add GeoJSON data and markers to the map
for index, row in df.iterrows():
    geojson_data = row["polygon_farm_footprint_geojson"]
    name = row["name"]
    latitude = row["latitude"]
    longitude = row["longitude"]

    # Create a GeoJson layer for the farm footprint
    farm_layer = folium.GeoJson(geojson_data)

    # Create a marker with a popup for additional information
    marker = folium.Marker(location=(latitude, longitude), popup=name)

    # Add both the GeoJson layer and the marker to the map
    farm_layer.add_to(map_object)
    marker.add_to(map_object)

# Add tile layer to map
tile = folium.TileLayer(
      tiles='https://server.arcgisonline.com/ArcGIS/rest/services/World_Imagery/MapServer/tile/{z}/{y}/{x}',
      attr='Esri',
      name='Esri Satellite',
      overlay=False,
      control=True
).add_to(map_object)

# Display the map
map_object

## Methodology

### Polygon area calculation
The Gauss's area formula, also known as the Shoelace formula, calculates the area of a simple polygon given its vertex coordinates. In mathematical form, the formula can be expressed as follows:


<br>
<img src = "Figure2.png", style="width: 275px";>

Where:
* n is the number of vertices of the polygon.
* (xi,yi) are the coordinates of the ith vertex.
* The absolute value ∣⋅∣ ensures that the calculated area is positive regardless of the order of the vertices.

This formula works for simple polygons, where the edges do not intersect except at the vertices and there are no holes in the polygon. The Shoelace formula gets its name from the visualisation of lacing up the polygon with shoelaces, forming triangles whose areas contribute to the total area of the polygon.

In [ ]:
import pyproj
import json

# Calculate area using the shoelace formula
def calculate_polygon_area(coordinates):
    n = len(coordinates)
    area = 0
    for i in range(n):
        x1, y1 = coordinates[i]
        x2, y2 = coordinates[(i + 1) % n]
        area += (x1 * y2 - x2 * y1)
    return abs(area) / 2

# Define the coordinate transformation (convert from EPSG:4326 to EPSG:3395, known as "World Mercator")
# Transform the geographical coordinates to a projected coordinate system (EPSG:3395) that uses meters as the unit of measurement
transformer = pyproj.Transformer.from_crs("EPSG:4326", "EPSG:3395", always_xy=True)

# Iterate over each row and calculate polygon areas
df["polygon_area_m2"] = None  # Initialize the area column
for index, row in df.iterrows():
    polygon_geojson = json.loads(row["polygon_farm_footprint_geojson"])
    polygon_coordinates = polygon_geojson["coordinates"][0]
    projected_coordinates = [transformer.transform(lon, lat) for lon, lat in polygon_coordinates]
    area = calculate_polygon_area(projected_coordinates)
    df.at[index, "polygon_area_m2"] = area

df

name  poi_id  \
0   Large Scale Solar (LSS) Power Plant Pekan (100...     NaN   
1   Large Scale Solar (LSS) Power Kuala Perlis - C...     NaN   
2   Large Scale Solar (LSS) Power Plant Padang Bes...     NaN   
3   Large Scale Solar (LSS) Power Plant Arau (5.2 MW)     NaN   
4   Large Scale Solar (LSS) Power Plant Ayer Keroh...     NaN   
5   Large Scale Solar (LSS) Power Plant Bidor (30 MW)     NaN   
6   Large Scale Solar (LSS) Power Plant Bukit Kayu...     NaN   
7   Large Scale Solar (LSS) Power Plant Bukit Mert...     NaN   
8   Large Scale Solar (LSS) Power Plant Bukit Sela...     NaN   
9   Large Scale Solar (LSS) Power Plant Bukit Sela...     NaN   
10  Large Scale Solar (LSS) Power Plant Bukit Sela...     NaN   
11  Large Scale Solar (LSS) Power Plant Chembong (...     NaN   
12  Large Scale Solar (LSS) Power Plant Chembong (...     NaN   
13  Large Scale Solar (LSS) Power Plant Dengkil (1...     NaN   
14  Large Scale Solar (LSS) Power Plant Gambang (6...     NaN   
15  Large Scale Solar (LSS) Power Plant Gurun (65 MW)     NaN   
16  Large Scale Solar (LSS) Power Plant Kampar (13...     NaN   
17  Large Scale Solar (LSS) Power Plant Kamunting ...     NaN   
18  Large Scale Solar (LSS) Power Plant Kemaman (1...     NaN   
19     Large Scale Solar (LSS) Power Plant KMB (5 MW)     NaN   
20  Large Scale Solar (LSS) Power Plant Kuala Keti...     NaN   
21  Large Scale Solar (LSS) Power Plant Nibong Teb...     NaN   
22  Large Scale Solar (LSS) Power Plant Pokok Sena...     NaN   
23  Large Scale Solar (LSS) Power Plant Rembau (60...     NaN   
24  Large Scale Solar (LSS) Power Plant Rembau One...     NaN   
25  Large Scale Solar (LSS) Power Plant Sepang (10...     NaN   
26  Large Scale Solar (LSS) Power Plant Sepang (50...     NaN   
27  Large Scale Solar (LSS) Power Plant Sungai Pet...     NaN   
28  Large Scale Solar (LSS) Power Plant Gebeng (10...     NaN   
29  Large Scale Solar (LSS) Power Plant Kerian (30...     NaN   
30  Large Scale Solar (LSS) Power Plant Quantum So...     NaN   
31  Large Scale Solar (LSS) Power Plant Pontian (5...     NaN   
32  Large Scale Solar (LSS) Power Plant Kerian (10...     NaN   

                   category  \
0   Large Scale Solar (LSS)   
1   Large Scale Solar (LSS)   
2   Large Scale Solar (LSS)   
3   Large Scale Solar (LSS)   
4   Large Scale Solar (LSS)   
5   Large Scale Solar (LSS)   
6   Large Scale Solar (LSS)   
7   Large Scale Solar (LSS)   
8   Large Scale Solar (LSS)   
9   Large Scale Solar (LSS)   
10  Large Scale Solar (LSS)   
11  Large Scale Solar (LSS)   
12  Large Scale Solar (LSS)   
13  Large Scale Solar (LSS)   
14  Large Scale Solar (LSS)   
15  Large Scale Solar (LSS)   
16  Large Scale Solar (LSS)   
17  Large Scale Solar (LSS)   
18  Large Scale Solar (LSS)   
19  Large Scale Solar (LSS)   
20  Large Scale Solar (LSS)   
21  Large Scale Solar (LSS)   
22  Large Scale Solar (LSS)   
23  Large Scale Solar (LSS)   
24  Large Scale Solar (LSS)   
25  Large Scale Solar (LSS)   
26  Large Scale Solar (LSS)   
27  Large Scale Solar (LSS)   
28  Large Scale Solar (LSS)   
29  Large Scale Solar (LSS)   
30  Large Scale Solar (LSS)   
31  Large Scale Solar (LSS)   
32  Large Scale Solar (LSS)   

                                              address   country  \
0   41, Jln Kampung Tanjung Selangor, 26600 Pekan,...  Malaysia   
1                          02000 Kuala Perlis, Perlis  Malaysia   
2             Titi Tinggi, 02100 Padang Besar, Perlis  Malaysia   
3             Unnamed Road, 02500, 02500 Arau, Perlis  Malaysia   
4   Kompleks Hijau Solar, Jalan Solar 1, Hang Tuah...  Malaysia   
5              Kampung Batu Empat, 35500 Bidor, Perak  Malaysia   
6                       06050 Bukit Kayu Hitam, Kedah  Malaysia   
7   No. 2887,Jalan Rozhan, OFF, Jalan Alma, 14000 ...  Malaysia   
8               5.648397284698055, 100.61736956380749  Malaysia   
9                     K17, 08010 Sungai Petani, Kedah  Malaysia   
10           Unnamed Road, 08000 Sungai Petani

### Potential CO2 storage by forest calculation

In areas where forests are most productive (i.e., moist tropical regions), they can sequester up to 11 tons of CO2 per hectare per year in above-ground biomass and additional carbon below ground.

In [ ]:
# Convert polygon_area_m2 to hectares (without creating a new column)
df["polygon_area_ha"] = df["polygon_area_m2"] / 10000  # 1 hectare = 10,000 square meters

# Calculate CO2 storage sequestration in tonnes
df["co2_storage_t"] = df["polygon_area_ha"] * 11  # 11 tonnes of CO2 sequestration per hectare per year

df

name  poi_id  \
0   Large Scale Solar (LSS) Power Plant Pekan (100...     NaN   
1   Large Scale Solar (LSS) Power Kuala Perlis - C...     NaN   
2   Large Scale Solar (LSS) Power Plant Padang Bes...     NaN   
3   Large Scale Solar (LSS) Power Plant Arau (5.2 MW)     NaN   
4   Large Scale Solar (LSS) Power Plant Ayer Keroh...     NaN   
5   Large Scale Solar (LSS) Power Plant Bidor (30 MW)     NaN   
6   Large Scale Solar (LSS) Power Plant Bukit Kayu...     NaN   
7   Large Scale Solar (LSS) Power Plant Bukit Mert...     NaN   
8   Large Scale Solar (LSS) Power Plant Bukit Sela...     NaN   
9   Large Scale Solar (LSS) Power Plant Bukit Sela...     NaN   
10  Large Scale Solar (LSS) Power Plant Bukit Sela...     NaN   
11  Large Scale Solar (LSS) Power Plant Chembong (...     NaN   
12  Large Scale Solar (LSS) Power Plant Chembong (...     NaN   
13  Large Scale Solar (LSS) Power Plant Dengkil (1...     NaN   
14  Large Scale Solar (LSS) Power Plant Gambang (6...     NaN   
15  Large Scale Solar (LSS) Power Plant Gurun (65 MW)     NaN   
16  Large Scale Solar (LSS) Power Plant Kampar (13...     NaN   
17  Large Scale Solar (LSS) Power Plant Kamunting ...     NaN   
18  Large Scale Solar (LSS) Power Plant Kemaman (1...     NaN   
19     Large Scale Solar (LSS) Power Plant KMB (5 MW)     NaN   
20  Large Scale Solar (LSS) Power Plant Kuala Keti...     NaN   
21  Large Scale Solar (LSS) Power Plant Nibong Teb...     NaN   
22  Large Scale Solar (LSS) Power Plant Pokok Sena...     NaN   
23  Large Scale Solar (LSS) Power Plant Rembau (60...     NaN   
24  Large Scale Solar (LSS) Power Plant Rembau One...     NaN   
25  Large Scale Solar (LSS) Power Plant Sepang (10...     NaN   
26  Large Scale Solar (LSS) Power Plant Sepang (50...     NaN   
27  Large Scale Solar (LSS) Power Plant Sungai Pet...     NaN   
28  Large Scale Solar (LSS) Power Plant Gebeng (10...     NaN   
29  Large Scale Solar (LSS) Power Plant Kerian (30...     NaN   
30  Large Scale Solar (LSS) Power Plant Quantum So...     NaN   
31  Large Scale Solar (LSS) Power Plant Pontian (5...     NaN   
32  Large Scale Solar (LSS) Power Plant Kerian (10...     NaN   

                   category  \
0   Large Scale Solar (LSS)   
1   Large Scale Solar (LSS)   
2   Large Scale Solar (LSS)   
3   Large Scale Solar (LSS)   
4   Large Scale Solar (LSS)   
5   Large Scale Solar (LSS)   
6   Large Scale Solar (LSS)   
7   Large Scale Solar (LSS)   
8   Large Scale Solar (LSS)   
9   Large Scale Solar (LSS)   
10  Large Scale Solar (LSS)   
11  Large Scale Solar (LSS)   
12  Large Scale Solar (LSS)   
13  Large Scale Solar (LSS)   
14  Large Scale Solar (LSS)   
15  Large Scale Solar (LSS)   
16  Large Scale Solar (LSS)   
17  Large Scale Solar (LSS)   
18  Large Scale Solar (LSS)   
19  Large Scale Solar (LSS)   
20  Large Scale Solar (LSS)   
21  Large Scale Solar (LSS)   
22  Large Scale Solar (LSS)   
23  Large Scale Solar (LSS)   
24  Large Scale Solar (LSS)   
25  Large Scale Solar (LSS)   
26  Large Scale Solar (LSS)   
27  Large Scale Solar (LSS)   
28  Large Scale Solar (LSS)   
29  Large Scale Solar (LSS)   
30  Large Scale Solar (LSS)   
31  Large Scale Solar (LSS)   
32  Large Scale Solar (LSS)   

                                              address   country  \
0   41, Jln Kampung Tanjung Selangor, 26600 Pekan,...  Malaysia   
1                          02000 Kuala Perlis, Perlis  Malaysia   
2             Titi Tinggi, 02100 Padang Besar, Perlis  Malaysia   
3             Unnamed Road, 02500, 02500 Arau, Perlis  Malaysia   
4   Kompleks Hijau Solar, Jalan Solar 1, Hang Tuah...  Malaysia   
5              Kampung Batu Empat, 35500 Bidor, Perak  Malaysia   
6                       06050 Bukit Kayu Hitam, Kedah  Malaysia   
7   No. 2887,Jalan Rozhan, OFF, Jalan Alma, 14000 ...  Malaysia   
8               5.648397284698055, 100.61736956380749  Malaysia   
9                     K17, 08010 Sungai Petani, Kedah  Malaysia   
10           Unnamed Road, 08000 Sungai Petani

### Potential CO2 reduction by generating electricity using solar panel calculation

*Annual Generation (kW) = System Capacity (kW) × Annual Solar Radiation (kW/m2) × Efficiency*

The daily average solar radiation received in Malaysia is about 4.21 kWh/m2 to 5.56 kWh/m2. While the annual solar radiation ranges from 1400 kWh/m2 to 1900 kWh/m2 resulting in an average of 1643 kWh/m2 every year.

Manufacturers rate solar panels by their efficiency, which ranges from around 15% to 20% of conversion of the sun's energy transformed into usable electricity.

In [ ]:
# Given the assumption values
annual_solar_radiation_kW = 1643
efficiency = 0.18

# Convert system capacity from MW to kW
df['capacity_kW'] = df['capacity_MW'] * 1000

# Calculate annual_generation_kW
df['annual_generation_kW'] = df['capacity_kW'] * annual_solar_radiation_kW * efficiency

*CO2 Emissions Without Solar Panels (kg) = Annual Electricity Consumption (kW) × Carbon Intensity (g/kW) / 1,000*

The most important figure in the energy balance of Malaysia is the total consumption of 150.06 billion kW of electric energy per year. Per capita have an average of 4,422kW. Malaysia could be self-sufficient with domestically produced energy.

Solar panels emit around 50g of CO2 per kWh produced in its first few years of operation. By the third year of having solar panels, most solar panels become carbon neutral. This is still roughly 20 times less than the carbon output of coal-powered electricity sources.

In [ ]:
# Given the assumption values
annual_electricity_consumption_kW = 150.06 * (10**9)
carbon_intensity1_kg_per_kW = 1

# Calculate CO2 Emissions Without Solar Panels
df['co2_without_solar_panels_kg'] = annual_electricity_consumption_kW * carbon_intensity1_kg_per_kW

*CO2 Emissions With Solar Panels (kg) = (Annual Electricity Consumption - Annual Generation) (kW) × Carbon Intensity (g/kW) / 1,000*

In [ ]:
# Given the assumption values
carbon_intensity2_kg_per_kW = 0.05

# Calculate CO2 Emissions with Solar Panels
df['co2_with_solar_panels_kg'] = (annual_electricity_consumption_kW - df['annual_generation_kW']) * carbon_intensity2_kg_per_kW

*Potential CO2 Reduction (kg) = CO2 Emissions Without Solar Panels - CO2 Emissions With Solar Panels*

In [ ]:
# Calculate Potential CO2 Reduction in kilograms
df['co2_reduction_kg'] = df['co2_without_solar_panels_kg'] - df['co2_with_solar_panels_kg']

# Convert to in tonnes
df['co2_reduction_t'] = df['co2_reduction_kg'] / 1000  # 1 kilogram = 1,000 tonnes

# Set the display format for floating-point numbers
pd.options.display.float_format = '{:.2f}'.format

df

name  poi_id  \
0   Large Scale Solar (LSS) Power Plant Pekan (100...     NaN   
1   Large Scale Solar (LSS) Power Kuala Perlis - C...     NaN   
2   Large Scale Solar (LSS) Power Plant Padang Bes...     NaN   
3   Large Scale Solar (LSS) Power Plant Arau (5.2 MW)     NaN   
4   Large Scale Solar (LSS) Power Plant Ayer Keroh...     NaN   
5   Large Scale Solar (LSS) Power Plant Bidor (30 MW)     NaN   
6   Large Scale Solar (LSS) Power Plant Bukit Kayu...     NaN   
7   Large Scale Solar (LSS) Power Plant Bukit Mert...     NaN   
8   Large Scale Solar (LSS) Power Plant Bukit Sela...     NaN   
9   Large Scale Solar (LSS) Power Plant Bukit Sela...     NaN   
10  Large Scale Solar (LSS) Power Plant Bukit Sela...     NaN   
11  Large Scale Solar (LSS) Power Plant Chembong (...     NaN   
12  Large Scale Solar (LSS) Power Plant Chembong (...     NaN   
13  Large Scale Solar (LSS) Power Plant Dengkil (1...     NaN   
14  Large Scale Solar (LSS) Power Plant Gambang (6...     NaN   
15  Large Scale Solar (LSS) Power Plant Gurun (65 MW)     NaN   
16  Large Scale Solar (LSS) Power Plant Kampar (13...     NaN   
17  Large Scale Solar (LSS) Power Plant Kamunting ...     NaN   
18  Large Scale Solar (LSS) Power Plant Kemaman (1...     NaN   
19     Large Scale Solar (LSS) Power Plant KMB (5 MW)     NaN   
20  Large Scale Solar (LSS) Power Plant Kuala Keti...     NaN   
21  Large Scale Solar (LSS) Power Plant Nibong Teb...     NaN   
22  Large Scale Solar (LSS) Power Plant Pokok Sena...     NaN   
23  Large Scale Solar (LSS) Power Plant Rembau (60...     NaN   
24  Large Scale Solar (LSS) Power Plant Rembau One...     NaN   
25  Large Scale Solar (LSS) Power Plant Sepang (10...     NaN   
26  Large Scale Solar (LSS) Power Plant Sepang (50...     NaN   
27  Large Scale Solar (LSS) Power Plant Sungai Pet...     NaN   
28  Large Scale Solar (LSS) Power Plant Gebeng (10...     NaN   
29  Large Scale Solar (LSS) Power Plant Kerian (30...     NaN   
30  Large Scale Solar (LSS) Power Plant Quantum So...     NaN   
31  Large Scale Solar (LSS) Power Plant Pontian (5...     NaN   
32  Large Scale Solar (LSS) Power Plant Kerian (10...     NaN   

                   category  \
0   Large Scale Solar (LSS)   
1   Large Scale Solar (LSS)   
2   Large Scale Solar (LSS)   
3   Large Scale Solar (LSS)   
4   Large Scale Solar (LSS)   
5   Large Scale Solar (LSS)   
6   Large Scale Solar (LSS)   
7   Large Scale Solar (LSS)   
8   Large Scale Solar (LSS)   
9   Large Scale Solar (LSS)   
10  Large Scale Solar (LSS)   
11  Large Scale Solar (LSS)   
12  Large Scale Solar (LSS)   
13  Large Scale Solar (LSS)   
14  Large Scale Solar (LSS)   
15  Large Scale Solar (LSS)   
16  Large Scale Solar (LSS)   
17  Large Scale Solar (LSS)   
18  Large Scale Solar (LSS)   
19  Large Scale Solar (LSS)   
20  Large Scale Solar (LSS)   
21  Large Scale Solar (LSS)   
22  Large Scale Solar (LSS)   
23  Large Scale Solar (LSS)   
24  Large Scale Solar (LSS)   
25  Large Scale Solar (LSS)   
26  Large Scale Solar (LSS)   
27  Large Scale Solar (LSS)   
28  Large Scale Solar (LSS)   
29  Large Scale Solar (LSS)   
30  Large Scale Solar (LSS)   
31  Large Scale Solar (LSS)   
32  Large Scale Solar (LSS)   

                                              address   country  \
0   41, Jln Kampung Tanjung Selangor, 26600 Pekan,...  Malaysia   
1                          02000 Kuala Perlis, Perlis  Malaysia   
2             Titi Tinggi, 02100 Padang Besar, Perlis  Malaysia   
3             Unnamed Road, 02500, 02500 Arau, Perlis  Malaysia   
4   Kompleks Hijau Solar, Jalan Solar 1, Hang Tuah...  Malaysia   
5              Kampung Batu Empat, 35500 Bidor, Perak  Malaysia   
6                       06050 Bukit Kayu Hitam, Kedah  Malaysia   
7   No. 2887,Jalan Rozhan, OFF, Jalan Alma, 14000 ...  Malaysia   
8               5.648397284698055, 100.61736956380749  Malaysia   
9                     K17, 08010 Sungai Petani, Kedah  Malaysia   
10           Unnamed Road, 08000 Sungai Petani

In [ ]:
df.to_csv('Solar_Farm_Malaysia_v0.2.0_Polygon - Sheet1.csv', index=False)

In [ ]:
# Calculate the average CO2 reduction per hectare
average_co2_reduction_per_ha = df['co2_reduction_t'].sum() / df['polygon_area_ha'].sum()

round(average_co2_reduction_per_ha, 2)

4621228.19

Common real life perspective to better understand the calculated value of total CO2 reduction:

- A typical passenger vehicle emits about 4.6 metric tons of carbon dioxide per year. If total CO2 reduction is 1000 tons, we could say it's equivalent to taking 217 cars off the road for a year.

- if CO2 reduction is 1,000 metric tons, it's also equivalent to planting and maintaining 13,000 trees for a year.

## Use Case

Regulatory compliance and reporting:
- AI can assist in ensuring that the solar farm adheres to environmental regulations and permits. It can track and report on parameters such as land use, emissions reduction and energy output.

Environmental impact assessment:
- AI models can assess the potential environmental impacts of the solar megawatt farm, considering factors like habitat disruption and visual impact. This information is useful for stakeholders and regulatory bodies.

Carbon sequestration estimation:
- AI can integrate data on forest type, age and location with carbon sequestration models to estimate the carbon storage capacity of the affected forests. It can also predict the carbon emissions resulting from deforestation.

Climate impact and carbon footprint:
- AI can analyse data related to energy consumption during solar panel production, transportation and construction to estimate the carbon footprint of the solar farm. It can also predict the carbon emissions reduced by the farm over its lifetime.

Economic impact assessment:
- AI can analyse economic data, including the cost of solar panel development, potential revenue from energy generation, and the economic value of ecosystem services provided by forests. It can model different scenarios to assess the economic implications.

## Recommendation

### Government

Promote low-carbon policies:
- Outline the aspiration for the nation to achieve net zero GHG emissions as early as 2050. Restricting the development of new coal power plants and encouraging the adoption of electric vehicles (EVs)

Implement regulatory framework:
- Establish clear regulations for solar development on forested lands, outlining specific conditions and criteria for approval. Strengthen penalties for non-compliance.

Promote reforestation and afforestation:
- Implement reforestation and afforestation programs to compensate for any forested lands that must be cleared for solar farms.

### Private sector

Solar farm efficiency and technology:
- Promote research and development in solar panel technology to increase efficiency, allowing for the generation of more energy from smaller land footprints like rooftop area. Incentivize the use of advanced solar technologies.

Biodiversity conservation:
- Collaborate with environmental organizations and experts to implement biodiversity conservation measures within solar farms, such as planting native vegetation and creating wildlife corridors.

Environmental certifications:
- Seek environmental certifications like LEED (Leadership in Energy and Environmental Design) to demonstrate a commitment to sustainability and responsible land use.